src/main/java/ch/uzh/ifi/hase/soprafs22/entity/Deck.java

```diff
@@ -1,53 +1,114 @@
 package ch.uzh.ifi.hase.soprafs22.entity;
 
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.HashSet;
-import java.util.List;
+import java.util.Map;
 import java.util.Set;
 
 import javax.persistence.*;
 
-import ch.uzh.ifi.hase.soprafs22.constant.GenerateRank;
+import org.springframework.web.client.RestTemplate;
 import ch.uzh.ifi.hase.soprafs22.constant.Rank;
 import ch.uzh.ifi.hase.soprafs22.constant.Suit;
+import ch.uzh.ifi.hase.soprafs22.rest.dto.DeckOfCardsAPI.*;
 
 @Entity
 public class Deck {
 
+    private final static String baseUrl = "https://deckofcardsapi.com/api/";
+    private final static String[] AVAILABLE_SUITS = new String[]{"S", "D", "C", "H"};
+    private final static String[] AVAILABLE_RANKS = new String[]{"2", "3", "4", "5", "6", "8", "9", "0", "Q", "K", "A"};
+
     @Id
     @GeneratedValue
     private Long id;
-    
-    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
-    private Set<Card> cards;
 
-    public Deck(){
-        this.cards = new HashSet<>();
-        newDeck();
-    }
+    @Column
+    private String deck_id;
+
+    @Transient
+    private RestTemplate restTemplate = new RestTemplate();
+
+    public Deck() {}
 
-    private void newDeck(){
-        this.cards.clear();
-        for (GenerateRank aRank : GenerateRank.values()) {
-            for (Suit aSuit : Suit.values()) {
-                this.cards.add(new Card(Rank.valueOf(aRank.toString()), aSuit));
+    public void initialize() {
+//        String url = baseUrl + "deck/new/shuffle/?deck_count=2&jokers_enabled=true";
+        StringBuilder url = new StringBuilder("https://deckofcardsapi.com/api/deck/new/shuffle/?cards=");
+        for(String suit : AVAILABLE_SUITS) {
+            for(String rank : AVAILABLE_RANKS) {
+                url.append(rank).append(suit).append(",");
             }
         }
-        // Dont think we have to shuffle cards anymore since set is unordered
+        url.deleteCharAt(url.length() - 1);
+        System.out.println(url.toString());
+        DeckDTO result = this.restTemplate.getForObject(url.toString(), DeckDTO.class);
+
+        this.deck_id = result.getDeck_id();
+        System.out.println("Deck initialized");
+    }
+
+    public void refillDeck() {
+        String url = String.format("%sdeck/%s/return/", baseUrl, this.deck_id);
+        this.restTemplate.getForObject(url, String.class);
+        System.out.println("Deck was refilled successfully");
```

Comment:
```suggestion
```

Reply:
Kept this one in because I wasnt sure if the deck needs to be refilled after every round.

Reply:
sure, it's probably a good idea. better be safe than sorry 😉